types/promise-nodeify/index.d.ts

```diff
@@ -0,0 +1,25 @@
+// Type definitions for promise-nodeify 3.0
+// Project: https://github.com/kevinoid/promise-nodeify
+// Definitions by: jgeth <https://github.com/jgeth>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace promiseNodeify;
+
+declare const promiseNodeify: {
+    <T>(
+        promise: Promise<T>,
+        callback: (error: Error | undefined, value: T | undefined) => any
+    ): Promise<T> | undefined;
+
+    delegated<T>(
+        promise: Promise<T>,
+        callback: (error: Error | undefined, value: T | undefined) => any
+    ): Promise<T> | undefined;
+
+    nodeifyThis<T>(
```

Comment:
This doesn't seem correct. The usage in the documentation is this:
```ts
Promise.prototype.nodeify = require('promise-nodify').nodeifyThis;
...
promise.nodeify(callback);
```

I think the definition you want is:
```ts
nodeifyThis<T>(
  this: Promise<T>,
  callback: (error: Error | undefined, value: T | undefined) => any
): Promise<T> | undefined;
```

Reply:
Good catch. It does use `this` but it's not in the method signature. Will update with correct signature.

Reply:
Still not correct. As I mentioned, `nodifyThis<T>` requires a `this`, so it would be better to change it to the following:
```suggestion
    nodeifyThis<T>(
      this: Promise<T>,
```

TypeScript uses a `this` parameter to denote the required `this` type for a function. Otherwise, it seems like `nodifyThis` can be called on its own with just a callback and produce a `Promise`, which is incorrect.